devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,554 @@
+		//fingerprint profileId: "0104", inClusters: "0000,0003,0201,0204,0405", outClusters: "0402,0405", manufacturer: "Viconics", model: "254-162", deviceJoinName: "VT8650xx"
+
+		// Schneider Electric SE8350 Low Voltage Fan Coil Unit (FCU) and Zone Control
+		// Raw Description 0A 0104 0301 00 0A 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0B 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-145", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller_Fan"
+
+		// Schneider Electric SE8650 Roof Top Unit Controller
+		// Raw Description 0A 0104 0301 00 09 0201 0405 0402 0406 0204 0000 0004 0003 0005 0A 0201 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-163", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller"
+	}
+}
+
+private getTHERMOSTAT_CLUSTER() { 0x0201 }
+private getTHERMOSTAT_UI_CONFIGURATION_CLUSTER() { 0x0204 }
+private getTEMPERATURE_DISPLAY_MODE() {0x0000}
+private getLOCAL_TEMPERATURE() {0x0000}
+private getCOOLING_SETPOINT() { 0x0011 }
+private getHEATING_SETPOINT() { 0x0012 }
+private getCOOLING_SETPOINT_UNOCCUPIED() { 0x0013 }
+private getHEATING_SETPOINT_UNOCCUPIED() { 0x0014 }
+private getOCCUPANCY() { 0x002 }
+private getCUSTOM_OCCUPANCY() {0x0650}
+private getCUSTOM_EFFECTIVE_OCCUPANCY() { 0x0c50 }
+private getCUSTOM_HUMIDITY() { 0x07a6 }
+private getCUSTOM_THERMOSTAT_MODE() { 0x0687 }
+private getCUSTOM_FAN_SPEED() { 0x0688 }
+private getCUSTOM_FAN_MODE() { 0x0698 }
+private getCUSTOM_THERMOSTAT_OPERATING_STATE() { 0x06BF }
+private getUNOCCUPIED_SETPOINT_CHILD_DEVICE_ID() {1}
+private getTHERMOSTAT_MODE_OFF() { 0x00 }
+private getTHERMOSTAT_MODE_AUTO() { 0x01 }
+private getTHERMOSTAT_MODE_COOL() { 0x02 }
+private getTHERMOSTAT_MODE_HEAT() { 0x03 }
+private getCUSTOM_FAN_MODE_ON() { 0x00 }
+private getCUSTOM_FAN_MODE_AUTO() { 0x01 }
+private getCUSTOM_FAN_MODE_CIRCULATE() { 0x02 }
+private getOPERATING_STATE_IDLE() { 0x00 }
+private getOPERATING_STATE_COOLING() { 0x01 }
+private getOPERATING_STATE_HEATING() { 0x02 }
+private getOCCUPANCY_OCCUPIED() { 0x00 }
+private getOCCUPANCY_UNOCCUPIED() { 0x01 }
+private getFAN_SPEED_LOW() { 0x00 }
+private getFAN_SPEED_MEDIUM() { 0x01 }
+private getFAN_SPEED_HIGH() { 0x02 }
+private getFAN_SPEED_AUTO() { 0x03 }
+
+
+private getFAN_MODE_MAP() {
+	[
+		(CUSTOM_FAN_MODE_ON):"on",
+		(CUSTOM_FAN_MODE_AUTO):"auto",
+		(CUSTOM_FAN_MODE_CIRCULATE):"circulate"
+	]
+}
+
+private getTHERMOSTAT_FAN_MODE_ATTRIBUTE_ID_MAP() {
+	[
+		"on":  (CUSTOM_FAN_MODE_ON),
+		"auto": (CUSTOM_FAN_MODE_AUTO),
+		"circulate": (CUSTOM_FAN_MODE_CIRCULATE)
+	]
+}
+
+private getTHERMOSTAT_MODE_MAP() {
+	[
+		(THERMOSTAT_MODE_OFF):"off",
+		(THERMOSTAT_MODE_AUTO):"auto",
+		(THERMOSTAT_MODE_COOL):"cool",
+		(THERMOSTAT_MODE_HEAT):"heat",
+		"04":"heat"
```

Comment:
is this emergency heat?

Reply:
I'm still not sure about that one. Viconics/Schneider uses this value (i saw it in their very old dths), but it's not mentioned in their manual.
I left this value intentionally (for testing purposes), but it will be removed probably.

Reply:
did you still want to remove it? It's still a string and I think all the other indices are not.

Reply:
I performed few tests and the device consumed 0x04 and 0x05 values, but after a while it switched to previous mode. 
Additionally: values >= 0x06 were rejected (device didn't even report setting them).
Anyway,  i removed "04" already.